Server/intake.php

```diff
@@ -170,32 +170,48 @@
     // Loot
     if (!empty($_POST['loot']) && $hunt_id > 0) {
         $loot_array = [];
-        $gold_array = [15, 47, 106, 138, 191, 194, 210, 226, 227, 260, 261, 262, 264, 265];
         foreach ($_POST['loot'] as $loot_item) {
-            $loot_item['amount'] = str_replace(",", "", $loot_item['amount']);
+
+            // Amount checks
             if (!is_numeric($loot_item['amount']) || $loot_item['amount'] < 1) {
                 continue;
             }
 
+            // Lucky checks
             $lucky = null;
-            if (!empty($loot_item['lucky'])) {
-                $lucky = $loot_item['lucky'] === 'true' ? 1 : 0;
-            }
+            if (!empty($loot_item['lucky'])) { $lucky = $loot_item['lucky'] === 'true' ? 1 : 0; }
 
-            $query = $pdo->prepare("SELECT id FROM loot WHERE name LIKE ?");
-            $query->execute(array($loot_item['name']));
-            $loot_id = $query->fetchColumn();
+            // Hitgrab item id checks
+            $hg_item_id = null;
+            if (is_numeric($loot_item['id'])) { $hg_item_id = $loot_item['id']; }
 
-            if (in_array($loot_id, $gold_array)) { // Blocking gold, there could be multiples of it, and not accurate
-                continue;
-            }
+            // Plural name checks
+            $plural_name = null;
+            if (!empty($loot_item['plural_name'])) { $plural_name = $loot_item['plural_name']; }
 
+            // Single name checks
+            $single_name = null;
+            if (!empty($loot_item['name'])) { $single_name = $loot_item['name']; }
+
+            // Get mhct id
+            $query = $pdo->prepare("SELECT MIN(id) FROM loot WHERE name LIKE ? OR name LIKE ? OR hg_item_id = ?");
+            $query->execute(array($single_name, $plural_name, $hg_item_id));
+            $loot_id = $query->fetchColumn();
+
+            // If no mhct id found, add the item to the database
             if (!$loot_id) {
-                $query = $pdo->prepare('INSERT INTO loot (name) VALUES (?)');
-                $query->execute(array($loot_item['name']));
+                $query = $pdo->prepare('INSERT INTO loot (name, hg_item_id, plural_name) VALUES (?, ?, ?)');
+                $query->execute(array($single_name, $hg_item_id, $plural_name));
                 $loot_id = $pdo->lastInsertId();
+
+            // If mhct id exists, update fields if neeeded
+            } else {
+                $query = $pdo->prepare('UPDATE loot SET name = COALESCE(name, ?), hg_item_id = COALESCE(hg_item_id, ?),
+                  plural_name = COALESCE(plural_name, ?) WHERE id = ?');
```

Comment:
It might be nicer to our database if we added ` AND (name is null OR hg_item_id is null OR plural_name is null)` so we don't keep updating values to their original values.

Reply:
mysql checks it automatically if the new value is same as old, and i reversed the order of things in coalesce to update items when they change in game.